tests/lib/Security/SecureRandomTest.php

```diff
@@ -81,4 +81,56 @@ public function testInvalidLengths($length): void {
 		$generator = $this->rng;
 		$generator->generate($length);
 	}
+
+	public static function invalidCharProviders(): array {
+		return [
+			'invalid_too_short' => ['abc'],
+			'invalid_duplicates' => ['aabcd'],
+			'invalid_non_ascii' => ["abcd\xf0"],
+		];
+	}
+
+	/**
+	 * @dataProvider invalidCharProviders
+	 */
```

Comment:
```suggestion
	#[DataProvider('invalidCharProviders')
```

(and add the class import)